### PR TITLE
LPS-158083

### PR DESCRIPTION
**Pull request by magjed4289**
https://issues.liferay.com/browse/LPS-158083

### CODE REVIEW

**Comment by magjed4289** [comment]
ci:test:frontend:nocompile

**Comment by magjed4289** [comment]
ci:test:frontend:nocompile

**Comment by susanchen3** [comment]
rerunning because CI was [restarted](https://liferay.slack.com/archives/C19PWQ74J/p1658262924148779?thread_ts=1658257324.051899&cid=C19PWQ74J)

**Comment by susanchen3** [comment]
ci:test:frontend:nocompile

**Comment by magjed4289** [comment]
ci:test:frontend:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase [hunk 1]
```diff
@@ -136,6 +136,64 @@ definition {
...
+			Type(
+				locator1 = "NavBar#APPLICATION_SEARCH_FIELD",
+				value1 = "ES Element");
+
+			Portlet._clickAddPortlet(portletName = "ES Element");
+
+			Refresh();
+		}
+
+		task ("Assert the src value of the element is correct") {
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase:187]: Then remote app src is rendered with non-root context.

**Review by john-co — APPROVED** [review]
Looks good, just some nitpicks at Given When Then syntax

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase [hunk 2]
```diff
@@ -136,6 +136,64 @@ definition {
...
 	}
 
+	@description = "LPS-158083 Remote App using non-root portal context"
+	@priority = "4"
+	test CustomElementCanBeCreatedInNonRootContext {
+		property app.server.types = "tomcat";
+		property database.types = "mysql";
+		property environment.acceptance = "true";
+		property portal.acceptance = "true";
+		property portal.context = "liferay";
```
[the remark this answers is not in the document]
Comment (magjed4289) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase:146]: I can try, but I guess properties run when setting up the environment before the test tasks are run.

**Comment by magjed4289** [comment]
ci:test:frontend:nocompile

**Comment by magjed4289** [comment]
ci:stop:frontend

**Comment by magjed4289** [comment]
ci:stop:test:frontend

**Comment by magjed4289** [comment]
ci:stop

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase (hunk 2 above)
[the remark this answers is not in the document]
Comment (magjed4289) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase:146]: OK I will be just checking the portalURL and asserting it has "liferay" :D

**Comment by magjed4289** [comment]
@john-co changes are merged but the commits that are solving the issue are not yet merged into master, so we need to wait until they are. If you find a moment during your shift, you can check run on your updated local master `git log --grep=LPS-158083`, if any commit with this LPS is on the master branch, we should be good to go with running `ci:test:frontend:nocompile` again.

**Comment by john-co** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
ci:test:frontend

**Comment by john-co** [comment]
ci:test:sf

**Comment by john-co** [comment]
ci:forward

**Comment by john-co** [comment]
Forwarding, even though the test fails it is a [positive positive test](https://liferay.slack.com/archives/CNBG06JS3/p1658516791216549?thread_ts=1657634685.915459&cid=CNBG06JS3) (even though there was code merged meant to fix this, the behavior is still broken). It caught the failure it was supposed to and we should use the test in follow up dev fix to validate it.

**Comment by john-co** [comment]
cc/ @izaera @dsanz , heads up since it looks like the [non-root context is still broken](https://liferay.slack.com/archives/CNBG06JS3/p1658516791216549?thread_ts=1657634685.915459&cid=CNBG06JS3). On the follow up fix, we can run RemoteApps#CustomElementCanBeCreatedInNonRootContext as an automation test to validate the fix. 

Easy way to check
1. Run `ci:test:frontend` and check that RemoteApps#CustomElementCanBeCreatedInNonRootContext is not listed under common failures (NOT under unique failures since test will be failing due to existing issue).

https://user-images.githubusercontent.com/35235266/180513701-cff7ef99-1ada-4d41-974a-6243ff5c6779.mp4

**Pull request closed without merging**